keel-core/src/main/kotlin/com/netflix/spinnaker/keel/veto/Veto.kt

```diff
@@ -28,34 +28,34 @@ import com.netflix.spinnaker.keel.api.Resource
 interface Veto {
 
   /**
-   * The name of the veto
+   * The name of the veto.
    */
   fun name(): String = javaClass.simpleName
 
   /**
-   * Check whether the resource can be checked according to this veto
+   * Check whether the resource can be checked according to this veto.
    */
-  fun check(resource: Resource<*>): VetoResponse
+  suspend fun check(resource: Resource<*>): VetoResponse
 
   /**
-   * The message format a veto accepts
+   * The message format a veto accepts.
    */
-  fun messageFormat(): Map<String, Any>
+  fun messageFormat(): Map<String, Any> = TODO("not implemented")
```

Comment:
nice

Reply:
I don't love the existence of these methods. They seem to be "optional" for implementations which tells me we're missing a layer of abstraction. I didn't want to change too much at once though.